lib/api/BackbeatAPI.js

```
@@ -615,6 +622,156 @@ class BackbeatAPI {
+            async.waterfall([
+                done => this._redisClient.batch([['get', key]], (err, res) => {
+                    if (err) {
+                        return done(err);
+                    }
+                    const [cmdErr, sourceRole] = res[0];
+                    if (cmdErr) {
+                        return next(cmdErr);
+                    }
+                    if (!sourceRole) {
+                        // error occurred, skip for next time
+                        return next(errors.InternalError);
+                    }
+                    const entry = new ObjectFailureEntry(key, sourceRole);
+                    return done(null, entry);
+                }),
+                (entry, done) => {
+                    if (!entry) {
+                        return done();
+                    }
+                    return this._getObjectQueueEntry(entry,
+                    (err, queueEntry) => {
+                        if (err) {
+                            // error occurred, skip for next time
+                            return next(errors.InternalError);
+                        }
+                        const contentLength = queueEntry.getContentLength();
+                        return this._deleteFailedCRRKey(key, err => {
+                            if (err) {
+                                // error occurred, skip for next time
+                                return next(errors.InternalError);
+                            }
+                            return done(null, contentLength);
+                        });
+                    });
+                },
+            ], (err, bytes) => {
+                if (err) {
+                    this._logger.debug('failed to delete duplicate object ' +
+                    'fail key', {
+                        method: 'BackbeatAPI._normalizeFailMetrics',
+                        key,
+                    });
+                    return next();
```

I think we want to return the error here if there is one.